### DISCUSSION TITLE
Is it possible to create webhooks using dynamic urls?

### DISCUSSION THREAD

**Discussion by gaetanodegennaro** [spatie/laravel-webhook-client#86] [Q&A] [answered]
With this Laravel dependency is possible to create dynamic webhooks?
I need to handle this kind of requests: mysite.com/webhook/{dynamicId}
Thank you!

**Comment by DavidPower** [comment, marked as answer]
I'm still hunting for a good solution to this question.

Were you able to find one?

**Reply by gaetanodegennaro** [reply]
Yes, you simply should use in you routes file something like Route::webhooks('/webhook/completed/{id}', 'webhook/completed');

**Reply by DavidPower** [reply]
I appreciate your reply. But this doesn't work for me. It consistently generates a "419 unknown status" error. And when I attempt to pick off Route::current()->parameter('id') in my ProcessWebhookJob, it's always null. 

I'm curious how you got it to work.

**Reply by DavidPower** [reply]
Back with an update. Here's how to make this method work. (NB: Step 2 was key for me).

1. Define a parameter in your route definition (i.e. web.php) in the following format: Route::webhooks('mypath/{id}', 'config-name');
2. IMPORTANT: Update VerifyCsrfToken.php to include a wildcard '*' at the end of your excluded route as follows: 
protected $except = [
        //
        'mypath/*',
    ];
3. Access the parameter in your ProcessWebhookJob.php as follows: $id = Route::current()->parameter('id');

Many thanks to @gaetanodegennaro for the initial clues.
